src/current/v23.1/connection-pooling.md

```diff
@@ -25,37 +25,149 @@ Each time an application reads or writes data, it will request one of the connec
 
-Cockroach Labs performed lab testing of various customer workloads and found no improvement in scalability beyond:
+### Validate connections in a pool
+
+After a connection pool initializes connections to CockroachDB clusters, those connections can occasionally break. This could be due to changes in the cluster topography, or rolling upgrades and restarts, or network disruptions.
+
+Validating connections is typically handled automatically by the connection pool. For example, in HikariCP the connection is validated whenever you request a connection from the pool, and the `keepaliveTime` property allows you to configure an interval to periodically check if the connections in the pool are valid. Whatever connection pool you use, make sure connection validation is enabled when running your application.
+
+#### Implement connection retry logic in your application
+
+To be resilient to connection closures, your application should use a retry loop to reissue transactions that were open when a connection was closed. For details, see [Connection retry loop]({% link {{ page.version.version }}/node-shutdown.md %}#connection-retry-loop).
+
+<section class="filter-content" markdown="1" data-scope="dedicated selfhosted">
+
+If you cannot tolerate connection errors during node drain, you can change the `server.shutdown.connection_wait` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) to allow SQL client connections to gracefully close before CockroachDB forcibly closes them. For guidance, see [Node Shutdown]({% link {{ page.version.version }}/node-shutdown.md %}#server-shutdown-connection_wait).
+
+</section>
+
+## Size connection pools
+
+<div class="filters clearfix">
+  <button class="filter-button page-level" data-scope="serverless"><strong>{{ site.data.products.serverless }}</strong></button>
+  <button class="filter-button page-level" data-scope="dedicated"><strong>{{ site.data.products.dedicated }}</strong></button>
+  <button class="filter-button page-level" data-scope="selfhosted"><strong>{{ site.data.products.core }}</strong></button>
+</div>
+
+If your connection pool is properly configured, the total number of connections to your cluster should be at least 100 times larger than the number of new connections per second to your cluster.
+
+Idle connections in CockroachDB do not consume many resources compared to PostgreSQL. Unlike PostgreSQL, which has a hard limit of 5000 connections, CockroachDB can safely support tens of thousands of connections.
+
+<section class="filter-content" markdown="1" data-scope="serverless">
+
+In {{ site.data.products.serverless }} clusters, the purpose of the connection pool is to avoid establishing a new SQL connection each time a query is run. You do not need to set a maximum number of simultaneous connections when configuring your connection pool. {{ site.data.products.serverless }} clusters will automatically scale up to meet the demand from multiple simultaneous client connections.
```

Comment:
> You do not need to set a maximum number of simultaneous connections when configuring your connection pool.

This contradicts the advice above where we do say to configure max conns to be > max concurrent queries. It also contradicts advice for Serverless functions, where we say that max conns = 1 in most cases.

This gets into the discussion above around "Set the maximum number of open connections"; maybe we're just forced to give different advice for different products.

Reply:
I vote for removing this paragraph. It seems unlikely to me that we'd have a customer suffer from setting a max pool size for a serverless cluster, even though they don't need to.